schema/edition_set.js

```diff
@@ -30,6 +33,33 @@ const EditionSetType = new GraphQLObjectType({
         const fallback = forsale ? "Available" : "Not for Sale"
         return !isEmpty(price) ? price : fallback
       },
+      deprecationReason: "Prefer to use `sale_message`.",
+    },
+    sale_message: {
+      type: GraphQLString,
+      resolve: ({ availability, availability_hidden, price, forsale }) => {
+        // Don't display anything if availability is hidden.
+        if (availability_hidden) {
+          return null
+        }
+
+        // If it's a supported availability, just return it (capitalized).
+        if (includes(EditionSetAvailabilities, availability)) {
+          return capitalizeFirstCharacter(availability)
+        }
+
+        // If there's a price string, just return it.
+        if (!isEmpty(price)) {
+          return price
```

Comment:
And Price here already account for range and price hidden options?

Reply:
Yup! It's actually `display_price` in gravity.